core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManager.java

```diff
@@ -235,14 +242,40 @@ void checkMaxEnvironmentRam(WorkspaceConfig config) throws LimitExceededExceptio
                                                  .mapToInt(machineCfg -> machineCfg.getLimits().getRam())
                                                  .sum();
             if (workspaceRam > maxRamPerEnv) {
-                throw new LimitExceededException(format("The maximum RAM per workspace is set to '%dmb' and you requested '%dmb'. " +
+                throw new LimitExceededException(format("The maximum RAM per workspace is set to '%s' and you requested '%s'. " +
                                                         "This value is set by your admin with the 'limits.workspace.env.ram' property",
-                                                        maxRamPerEnv,
-                                                        workspaceRam));
+                                                        parseMegabyteSizeToString(maxRamPerEnv, "mb"),
+                                                        parseMegabyteSizeToString(workspaceRam, "mb")));
             }
         }
     }
 
+    private String parseMegabyteSizeToString(long ramSizeMB, String suffix) {
```

Comment:
Maybe it is better to use https://github.com/eclipse/che/blob/master/core/commons/che-core-commons-lang/src/main/java/org/eclipse/che/commons/lang/Size.java#L22-22 here.